cmake/GNU.cmake

```diff
@@ -0,0 +1,58 @@
+
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -ffree-line-length-none -fno-range-check -fbacktrace")
+
+if(DEBUG)
+    message("DEBUG is       ENABLED")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Wall -O0 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -O0 -ggdb")
+else()
+    message("DEBUG is       disabled (optimized build)")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -fno-range-check")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
+endif()
+
+
+if(REPRO)
+    message("REPRO is       ENABLED")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -ggdb")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
+else()
+    message("REPRO is       disabled")
+endif()
+
```

Comment:
I would also add a mechanism to either abort if both REPRO and DEBUG are requested, or make one of the two superior (I suggest DEBUG). In the latter case, issue a warning that both REPRO and DEBUG where requested, and that DEBUG will be used (and REPRO disabled).